dbms/src/Flash/Mpp/ExchangeReceiver.cpp

```diff
@@ -193,13 +196,23 @@ class AsyncRequestHandler : public UnaryCallback<bool>
     {
         switch (stage)
         {
+        case AsyncRequestStage::WAIT_RETRY:
+            start();
+            break;
         case AsyncRequestStage::WAIT_MAKE_READER:
         {
             // Use lock to ensure reader is created already in reactor thread
             std::unique_lock lock(mu);
             if (!ok)
+            {
                 reader.reset();
-            notifyReactor();
+                LOG_WARNING(log, "MakeReader fail. retry time: {}", retry_times);
+                retryOrDone("Exchange receiver meet error : send async stream request fail");
+            }
+            else
+            {
+                notifyReactor();
```

Comment:
why not just set the state to `wait_batch_read` and start the read?

Reply:
Done